src/librustc_mir/hair/pattern/check_match.rs

```diff
@@ -225,7 +225,15 @@ impl<'a, 'tcx> MatchVisitor<'a, 'tcx> {
                 let scrutinee_is_uninhabited = if self.tcx.features().exhaustive_patterns {
                     self.tcx.is_ty_uninhabited_from(module, pat_ty)
                 } else {
-                    self.conservative_is_uninhabited(pat_ty)
+                    if self.tcx.features().exhaustive_patterns {
+                        pat_ty.conservative_is_uninhabited()
```

Comment:
Isn't this checked 2 lines before, preventing this from being reached?

Reply:
i.e. `match` uses `is_ty_uninhabited_from` instead of `conservative_is_uninhabited`.

Reply:
Ah, this is where using `exhaustive_patterns` for two different things causes problems. I'll default to the old check here instead.